esmf_regrid/schemes.py

```diff
@@ -572,32 +1051,176 @@ def __call__(self, cube):
             A :class:`~iris.cube.Cube` defined with the horizontal dimensions of the target
             and the other dimensions from this :class:`~iris.cube.Cube`. The data values of
             this :class:`~iris.cube.Cube` will be converted to values on the new grid using
-            area-weighted regridding via :mod:`esmpy` generated weights.
+            regridding via :mod:`esmpy` generated weights.
 
         """
-        src_x, src_y = (_get_coord(cube, "x"), _get_coord(cube, "y"))
-
-        # Check the source grid matches that used in initialisation
-        if self.src_grid != (src_x, src_y):
-            raise ValueError(
-                "The given cube is not defined on the same "
-                "source grid as this regridder."
-            )
+        if cube.mesh is not None:
+            # TODO: replace temporary hack when iris issues are sorted.
+            # Ignore differences in var_name that might be caused by saving.
+            # TODO: uncomment this when iris issue with masked array comparison is sorted.
+            # self_mesh = copy.deepcopy(self._src.mesh)
+            # self_mesh.var_name = mesh.var_name
+            # for self_coord, other_coord in zip(self_mesh.all_coords, mesh.all_coords):
+            #     if self_coord is not None:
+            #         self_coord.var_name = other_coord.var_name
+            # for self_con, other_con in zip(
+            #     self_mesh.all_connectivities, mesh.all_connectivities
+            # ):
+            #     if self_con is not None:
+            #         self_con.var_name = other_con.var_name
+            # if self_mesh != mesh:
+            #     raise ValueError(
+            #         "The given cube is not defined on the same "
+            #         "source mesh as this regridder."
+            #     )
+            dims = [cube.mesh_dim()]
 
-        if len(src_x.shape) == 1:
-            grid_x_dim = cube.coord_dims(src_x)[0]
-            grid_y_dim = cube.coord_dims(src_y)[0]
         else:
-            grid_y_dim, grid_x_dim = cube.coord_dims(src_x)
+            new_src_x, new_src_y = (_get_coord(cube, "x"), _get_coord(cube, "y"))
+
+            # Check the source grid matches that used in initialisation
+            for saved_coord, new_coord in zip(self._src, (new_src_x, new_src_y)):
+                # Ignore differences in var_name that might be caused by saving.
+                saved_coord = copy.deepcopy(saved_coord)
+                saved_coord.var_name = new_coord.var_name
+                if saved_coord != new_coord:
+                    raise ValueError(
+                        "The given cube is not defined on the same "
+                        "source grid as this regridder."
+                    )
+
+            if len(new_src_x.shape) == 1:
+                dims = [cube.coord_dims(new_src_x)[0], cube.coord_dims(new_src_y)[0]]
+            else:
+                dims = cube.coord_dims(new_src_x)[::-1]
```

Comment:
If this is consistent with the other cases of `[::-1]` then it would be good to have the same comment here too. If there is a different reason this time then a different comment would be helpful.